pytensor/tensor/subtensor.py

```diff
@@ -162,13 +187,21 @@ def as_index_literal(
     ------
     NotScalarConstantError
     """
-    if idx == np.newaxis or isinstance(getattr(idx, "type", None), NoneTypeT):
-        return np.newaxis
+    if isinstance(idx, int | np.integer):
+        return idx
+
+    # NOTE: np.newaxis is None
+    idxtype = getattr(idx, "type", None)
```

Comment:
All `Variable` should have a type, so you can check `if isinstance(idx, Variable): idxtype = idx.type`

Reply:
I see you used that below, I think it's more readable here as well?